test/common/config/config_provider_impl_test.cc

```diff
@@ -188,31 +191,38 @@ class DummyConfigProviderManager : public ConfigProviderManagerImplBase {
   // Envoy::Config::ConfigProviderManager
   ConfigProviderPtr
   createStaticConfigProvider(const Protobuf::Message& config_proto,
-                             Server::Configuration::FactoryContext& factory_context) override {
+                             Server::Configuration::FactoryContext& factory_context,
+                             const Envoy::Config::ConfigProviderManager::OptionalArg&) override {
     return std::make_unique<StaticDummyConfigProvider>(
         dynamic_cast<const test::common::config::DummyConfig&>(config_proto), factory_context,
         *this);
   }
+  ConfigProviderPtr
+  createStaticConfigProvider(std::vector<std::unique_ptr<const Protobuf::Message>>&&,
+                             Server::Configuration::FactoryContext&, const OptionalArg&) override {
+    ASSERT(false || "this provider does not expect multiple config protos");
```

Comment:
`ASSERT_FALSE(true) << "some stuff"?` Or something like that (I think gtest has some dedicated predicate for this somewhere).

Reply:
This virtual function does not return an AssertionResult, so I can't use the gtest ASSERT_*s.

Reply:
Can you do `ASSERT(false, "some message")` then for better readability?

Reply:
Done.